.github/workflows/regression-testing/regressions-util.py

```diff
@@ -0,0 +1,494 @@
+from typing import Tuple
+
+import os
+import sys
+import argparse
+import numpy as np
+import pandas as pd
+
+CLASSIFICATION_METRICS = ['acc', 'auc', 'balacc', 'logloss']
+REGRESSION_METRICS = ['mae', 'r2', 'rmse']
+METRICS = CLASSIFICATION_METRICS + REGRESSION_METRICS
+TARGET_YAML_FILE = 'target.yaml'
+
+def _get_mean_results_across_folds(df) -> pd.DataFrame:
+    """ Returns a dataframe with the task, id, metric and the mean values
+        across folds
+
+        [idx: 'task', 'id', 'metric', ... mean metrics across folds ...]
+    """
+    # Get the information about id and metric, only need info from first fold
+
+    # [idx: task, id, metric]
+    df_info = df[df['fold'] == 0][['task', 'id', 'metric']].set_index('task')
+
+    # [idx: task, ... mean metrics across folds ...]
+    available_metrics = list(set(METRICS).intersection(set(df.columns)))
+    df_means = df[['task'] + available_metrics].groupby(['task']).mean()
+
+    return df_info.join(df_means)
+
+def generate_framework_def(
+    userdir: str,
+    username: str,
+    branch: str,  # Not used in this setup but perhaps in a different one
+    commit: str,
+):
+    """ Creates a framework definition to run an autosklearn repo.
+
+    Technically we only use the commit to pull the targeted version but for
+    naming consistency, we need to know the branch too.
+
+    Parameters
+    ----------
+    userdir: str
+        The path to where the framework definition should be placed
+
+    username: str
+        The username of the auto-sklearn fork owner to pull from, used as
+
+        * ssh: git@github.com:{username}/auto-sklearn
+
+    branch: str
+        Which branch of the fork to pull from
+        e.g.
+            development
+
+    commit: str
+        A full commit hash to use
+        e.g.
+            8b474a437ce980bd0909db59141b40d56f6d5688
+            or
+            #8b474a437ce980bd0909db59141b40d56f6d5688
+    """
+    assert len(commit) == 41 and commit[0] == '#' or len(commit) == 40, \
+        "Not a commit hash"
+
+    # automlbenchmark requires the '#' to identify it's a commit rather than
+    # a branch being targeted
+    if commit[0] != '#':
+        commit = '#' + commit
+
+    repo = f'git@github.com:{username}/auto-sklearn.git'
+
+    # Create the framework file
+    lines = '\n'.join([
+        f"---",
+        f"autosklearn_targeted:",
+        f"  extends: autosklearn",
+        f"  version: '{commit}'",
+        f"  repo: '{repo}'"
+    ])
+
+    filepath = os.path.join(userdir, 'frameworks.yaml')
+    with open(filepath, 'w') as f:
+        f.writelines(lines)
+
+def create_comparison(
```

Comment:
It turns out that those python files are almost the same - could we use a single python file where the differences are passed in as command line arguments?

Reply:
Originally I had a folder that was separate for `generate-baselines.yml` and `regression-testing.yml` but they share too much so they were merged into one folder, I forgot to delete the old folder `./regression-testing`

Reply:
That makes a lot of sense and makes looking at this a lot easier :)